core/services/gasupdater/gas_updater.go

```diff
@@ -152,7 +152,11 @@ func (gu *gasUpdater) runLoop() {
 		case <-gu.ctx.Done():
 			return
 		case <-gu.mb.Notify():
-			head := gu.mb.Retrieve()
+			head, exists := gu.mb.Retrieve()
+			if !exists {
+				logger.Info("GasUpdater: no head to retrieve. It might have been skipped")
```

Comment:
Doesn't this indicate that it was already processed?

Reply:
yeah exactly -  all items were already picked (for now) and there was one excessive iteration of the loop